src/hotspot/share/gc/parallel/psParallelCompact.cpp

```diff
@@ -2614,9 +2619,11 @@ void PSParallelCompact::compact() {
   }
 
   {
-    // Update the deferred objects, if any.  Any compaction manager can be used.
     GCTraceTime(Trace, gc, phases) tm("Deferred Updates", &_gc_timer);
-    ParCompactionManager* cm = ParCompactionManager::manager_array(0);
+    // Update the deferred objects, if any. Any compaction manager can be used.
+    // However, since the current thread is VM thread, we use the rightful one
+    // to keep the verification logic happy.
```

Comment:
If using a different cm makes the verification logic unhappy, that suggests "Any compaction manager can be used" isn't exactly true.  Maybe something like "In principle, any compaction manager would do.  However, ..."